### PR TITLE
updates to standardise execution env use pages - docker

### PR DESCRIPTION
**Pull request by rosieyohannan**
# Description
Updating "Using ..." pages for all execution environments to standardise how this information is formatted and presented. This is the Docker PR, there will be a PR for each execution environment.

# Reasons
A link to a GitHub and/or JIRA issue (if applicable).
Otherwise, a brief sentence about why you made these changes.

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: https://circleci.com/docs/style/.

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [x] Break up walls of text by adding paragraph breaks.
- [x] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [x] Keep the title between 20 and 70 characters.
- [x] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [x] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [x] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [x] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_cci2/using-docker.md [hunk 1]
```diff
@@ -15,115 +15,91 @@ version:
...
+CircleCI maintains convenience images on Docker Hub for popular languages. See [the CircleCI Developer Hub](https://circleci.com/developer/images) for a complete list of image names and tags. 
 
-- Avoid using mutable tags like `latest` or `1` as the image version in your `config.yml file`. It is best practice to use precise image versions or digests, like `redis:3.2.7` or `redis@sha256:95f0c9434f37db0a4f...` as shown in the examples. Mutable tags often lead to unexpected changes in your job environment.  CircleCI cannot guarantee that mutable tags will return an up-to-date version of an image. You could specify `alpine:latest` and actually get a stale cache from a month ago.
+**Note**: If you need a Docker image that installs Docker and has Git, consider using `cimg/base:current`.
 
-- If you experience increases in your run times due to installing additional tools during execution, consider creating and using a custom-built image that comes with those tools pre-installed. See the [Using Custom-Built Docker Images]({{site.baseurl}}/2.0/custom-images/) page for more information.
+## Specifying Docker images
+{: Specifying Docker images }
```
Comment (rosieyohannan) [inline jekyll/_cci2/using-docker.md:36]:
```suggestion
{: #specifying-docker-images }
```

**Review by kbatuigas — CHANGES_REQUESTED** [review]
Just a couple comments!

File: jekyll/_cci2/using-docker.md [hunk 2]
```diff
@@ -15,115 +15,91 @@ version:
...
 
-In this example, all steps run in the container created by the first image listed under the `build` job.
-
-To make the transition easy, CircleCI maintains convenience images on Docker Hub for popular languages. See [Using Pre-Built CircleCI Docker Images]({{ site.baseurl }}/2.0/circleci-images/) for the complete list of names and tags. If you need a Docker image that installs Docker and has Git, consider using `cimg/base:current`.
-
-### Docker image best practices
-{: #docker-image-best-practices }
-
-- If you encounter problems with rate limits imposed by your registry provider, using [authenticated docker pulls]({{ site.baseurl }}/2.0/private-images/) may grant higher limits.
+A container is an instance of a specified Docker image. The first image listed in your configuration for a job is referred to as the _primary_ container image, and this is where all steps in the job will run. _Secondary_ containers can also be specified to run alongside for running services, such as, databases. If you are new to Docker, see the [Docker Overview documentation](https://docs.docker.com/engine/docker-overview/) for concepts.
```
Comment (rosieyohannan) [inline jekyll/_cci2/using-docker.md:29]:
```suggestion
A container is an instance of a specified Docker image. The first image listed in your configuration for a job is referred to as the _primary_ container image and this is where all steps in the job will run. _Secondary_ containers can also be specified to run alongside for running services, such as, databases. If you are new to Docker, see the [Docker Overview documentation](https://docs.docker.com/engine/docker-overview/) for concepts.
```

File: jekyll/_cci2/using-docker.md [hunk 3]
```diff
@@ -15,115 +15,91 @@ version:
...
 
-- CircleCI has partnered with Docker to ensure that our users can continue to access Docker Hub without rate limits. As of November 1st 2020, with few exceptions, you should not be impacted by any rate limits when pulling images from Docker Hub through CircleCI. However, these rate limits may go into effect for CircleCI users in the future. We encourage you to [add Docker Hub authentication]({{ site.baseurl }}/2.0/private-images/) to your CircleCI configuration and consider upgrading your Docker Hub plan, as appropriate, to prevent any impact from rate limits in the future.
+CircleCI maintains convenience images on Docker Hub for popular languages. See [the CircleCI Developer Hub](https://circleci.com/developer/images) for a complete list of image names and tags.
```
Comment (rosieyohannan) [inline jekyll/_cci2/using-docker.md:31]:
```suggestion
CircleCI maintains convenience images on Docker Hub for popular languages. See [the CircleCI Developer Hub](https://circleci.com/developer/images) for a complete list of image names and tags.
```

**Review by kbatuigas — APPROVED** [review]
🎉

**Pull request merged**